index.ts

```diff
@@ -196,26 +196,29 @@ function suiteClassCallback(target: SuiteCtor, context: TestFunctions) {
 
             // assert testName is truthy
             if (parameters) {
-                const nameForParameters = method[nameForParametersSymbol];
-                parameters.forEach((parameterOptions, i) => {
-                    const { mark, name, params } = parameterOptions;
-
-                    let parametersTestName = `${testName}_${i}`;
-                    if (name) {
-                        parametersTestName = name;
-                    } else if (nameForParameters) {
-                        parametersTestName = nameForParameters(params);
-                    }
-
-                    const shouldSkipParam = shouldSkip || (mark === Mark.skip);
-                    const shouldOnlyParam = shouldOnly || (mark === Mark.only);
-                    const shouldPendingParam = shouldPending || (mark === Mark.pending);
-
-                    const testFunc = ((shouldPendingParam || shouldSkipParam) && context.it.skip)
-                            || (shouldOnlyParam && context.it.only)
-                            || context.it;
-
-                    applyTestFunc(testFunc, parametersTestName, method, [params]);
+                // we make the parameterised test a child suite so we can late bind the parameterised tests
+                describe(testName, function() {
```

Comment:
Only this should be something like context.describe(...)

Reply:
will try that

Reply:
done